qiskit_superstaq/custom_gates.py

```diff
@@ -145,6 +145,45 @@ def __str__(self) -> str:
         return f"ZZSwapGate({args})"
 
 
+class StrippedCZGate(qiskit.circuit.Gate):
+    """The Stripped CZ Gate is the gate that is actually being performed by Hilbert,
+    which is just a regular CZ gate when phi = 0, and is corrected into a CZ
+    gate by RZ gates afterwards if phi is nonzero.
+    """
+
+    def __init__(self, phi: float) -> None:
+        super().__init__("strippedcz", 2, [phi])
+
+    def inverse(self) -> "StrippedCZGate":
+        return StrippedCZGate(-self.params[0])
+
+    def _define(self) -> None:
+        qc = qiskit.QuantumCircuit(2, name="strippedcz")
+        qc.rz(self.params[0], 0)
+        qc.rz(self.params[0], 1)
+        qc.cz(0, 1)
+        self.definition = qc
+
+    def __array__(self, dtype: Optional[type] = None) -> npt.NDArray[np.bool_]:
```

Comment:
(this is apparently incorrect throughout this file) the returned array has complex values, not bools. technically the return type should depend on the `dtype` argument, but this is a little complex to implement (and probably needs to be corrected a bunch of other places) so for this pr i think we should at least make the default case correct:
```suggestion
    def __array__(self, dtype: Optional[type] = None) -> npt.NDArray[np.complex_]:
```